Extensions/xkit_patches.js

```diff
@@ -2606,3 +2601,54 @@ XKit.toast = {
 		}, 5000);
 	}
 };
+
+// Cheat the headers we send
+XKit.tools.Nx_XHR = function(params) {
+	params.timestamp = new Date().getTime();
+	XKit.tools.add_function(function() {
+		var XHR = new XMLHttpRequest();
+		XHR.open(add_tag.method, add_tag.url, add_tag.async || false);
+		if (add_tag.json === true) {
+			XHR.setRequestHeader("Content-type", "application/json");
+		}
+		for (var x in add_tag.headers) {
+			XHR.setRequestHeader(x, add_tag.headers[x]);
+		}
+		function callback(result) {
+			window.postMessage({
+				response: {
+					status: XHR.status,
+					responseText: XHR.response
+				},
+				headers: XHR.getAllResponseHeaders().split("\r\n"),
+				timestamp: "xkit_" + add_tag.timestamp,
+				success: result
+			}, window.location.protocol + "//" + window.location.host);
+		}
+		XHR.onerror = function() { callback(false); };
+		XHR.onload = function() { callback(true); };
+		if (typeof add_tag.data !== "undefined") {
+			XHR.send(add_tag.data);
+		} else {
+			XHR.send();
+		}
+	}, true, params);
+	window.addEventListener("message", function(e) {
```

Comment:
This event listener should remove itself so that this code will leak less memory over time (come to think of it, if add_function could remove its script tag after execution this would leak no memory over time)